src/CommunityToolkit.Maui.MediaElement/Extensions/SubtitleExtensions.android.cs

```diff
@@ -0,0 +1,242 @@
+		}
+		catch (Exception ex)
+		{
+			System.Diagnostics.Trace.TraceError(ex.Message);
+			return;
+		}
+		cues = mediaElement.SubtitleUrl switch
+		{
+			var url when url.EndsWith("srt") => SrtParser.ParseSrtContent(vttContent),
+			var url when url.EndsWith("vtt") => VttParser.ParseVttContent(vttContent),
+			_ => throw new NotSupportedException("Unsupported subtitle format"),
+		};
+	}
+
+	/// <summary>
+	/// Starts the subtitle display.
+	/// </summary>
+	public void StartSubtitleDisplay()
+	{
+		ArgumentNullException.ThrowIfNull(subtitleView);
+		if(styledPlayerView.Parent is not ViewGroup parent)
+		{
+			System.Diagnostics.Trace.TraceError("StyledPlayerView parent is not a ViewGroup");
+			return;
+		}
+		dispatcher.Dispatch(() => parent.AddView(subtitleView));
+		timer = new System.Timers.Timer(1000);
+		timer.Elapsed += UpdateSubtitle;
+		timer.Start();
+	}
+
+	/// <summary>
+	/// Stops the subtitle timer.
+	/// </summary>
+	public void StopSubtitleDisplay()
+	{
+		if (timer is null || subtitleView is null)
+		{
+			return;
+		}
+		if (styledPlayerView.Parent is ViewGroup parent)
+		{
+			dispatcher.Dispatch(() => parent.RemoveView(subtitleView));
+		}
+		subtitleView.Text = string.Empty;
+		timer.Stop();
+		timer.Elapsed -= UpdateSubtitle;
+	}
+
+	void UpdateSubtitle(object? sender, System.Timers.ElapsedEventArgs e)
+	{
+		ArgumentNullException.ThrowIfNull(subtitleView);
+		ArgumentNullException.ThrowIfNull(mediaElement);
+		if (cues.Count == 0)
+		{
+			return;
+		}
+
+		var cue = cues.Find(c => c.StartTime <= mediaElement.Position && c.EndTime >= mediaElement.Position);
+		dispatcher.Dispatch(() =>
+		{
+			if (cue is not null)
+			{
+				subtitleView.FontFeatureSettings = !string.IsNullOrEmpty(mediaElement.SubtitleFont) ? mediaElement.SubtitleFont : default;
+				subtitleView.Text = cue.Text;
+				subtitleView.TextSize = (float)mediaElement.SubtitleFontSize;
+				subtitleView.Visibility = Android.Views.ViewStates.Visible;
+			}
+			else
+			{
+				subtitleView.Text = string.Empty;
+				subtitleView.Visibility = Android.Views.ViewStates.Gone;
+			}
+		});
+	}
+
+	void InitializeTextBlock()
+	{
+		var (currentActivity, _, _, _) = VerifyAndRetrieveCurrentWindowResources();
+		var activity = currentActivity;
+		subtitleView = new(activity.ApplicationContext)
+		{
+			Text = string.Empty,
+			HorizontalScrollBarEnabled = false,
+			VerticalScrollBarEnabled = false,
+			TextAlignment = Android.Views.TextAlignment.Center,
+			Visibility = Android.Views.ViewStates.Gone,
+			LayoutParameters = subtitleLayout
+		};
+		subtitleView.SetBackgroundColor(Android.Graphics.Color.Argb(150, 0, 0, 0));
+		subtitleView.SetTextColor(Android.Graphics.Color.White);
+		subtitleView.SetPaddingRelative(10, 10, 10, 20);
+	}
+
+	static (Activity CurrentActivity, Android.Views.Window CurrentWindow, Resources CurrentWindowResources, Configuration CurrentWindowConfiguration) VerifyAndRetrieveCurrentWindowResources()
```

Comment:
can we have a `readonly struct record` instead of ValueTuples?

Reply:
I can do that. BTW I copied this from android implementation for full screen. Do U want to change it there too in a separate PR?

Reply:
yeah, that would be great <3